cpp/net/connection_pool.cc

```diff
@@ -16,45 +17,92 @@ DEFINE_int32(url_fetcher_max_conn_per_host_port, 4,
 namespace cert_trans {
 namespace internal {
 
+ConnectionPool::Connection::Connection(evhtp_connection_t* conn,
+                                       HostPortPair&& other_end)
+    : conn_(CHECK_NOTNULL(conn)), other_end_(move(other_end)) {
+}
+
+
+const HostPortPair& ConnectionPool::Connection::other_end() const {
+  return other_end_;
+}
+
+
 ConnectionPool::ConnectionPool(libevent::Base* base)
     : base_(CHECK_NOTNULL(base)), cleanup_scheduled_(false) {
 }
 
 
-evhttp_connection_unique_ptr ConnectionPool::Get(const URL& url) {
+// static
+evhtp_res ConnectionPool::Connection::ConnectionClosedHook(
+    evhtp_connection_t* conn, void* arg) {
+  CHECK_NOTNULL(conn);
+  CHECK_NOTNULL(arg);
+  VLOG(1) << "Releasing connection " << conn;
```

Comment:
Is there anywhere else where this address will have been printed? As in, something that could be correlated with another log message? Maybe logging `c` would be more useful? I didn't fully check, not sure that gets logged either... If not, maybe logging the host:port (using `c`) could be *actually* useful? :wink: